pkg/store/bucket.go

```diff
@@ -1101,7 +1111,7 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 				}
 
 				mtx.Lock()
-				res = append(res, part)
+				res = append(res, newSortedSeriesSet(part, sortWithoutLabelSet))
```

Comment:
We also need to send labels at the end here. Basically every time we want to do a merge sort, replica labels need to be at the end of the individual series.